### PR TITLE
modules/aws/ami: Use implicit provider inheritance

### PR DESCRIPTION
**Pull request by wking**
Docs [here][1] say:

> This approach is recommended in the common case where only a single configuration is needed for each provider across the entire configuration.

I didn't know about implicit provider inheritance when I landed the module in #84.  We'll probably drop this module soon as we move to Red Hat CoreOS, but I worked up the change while looking for patterns to follow for a bootstrap module.

[1]: https://www.terraform.io/docs/modules/usage.html#implicit-provider-inheritance

### CODE REVIEW

**Comment by wking** [comment]
retest this please

**Comment by wking** [comment]
All green.  @abhinavdahiya, @yifan-gu, please take a look sometime tomorrow.

File: modules/aws/ami/README.md [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
-From the module directory:
+Set up a `main.tf` with:
+
+```hcl
+provider "aws" {
+  region = "us-east-1"
+}
+
+module "ami" {
+  source = "github.com/openshift/installer//modules/aws/ami"
```
Comment (abhinavdahiya) [inline modules/aws/ami/README.md]: `//` is that necessary?

Comment (wking) [reply modules/aws/ami/README.md]: > // is that necessary?

Yup.  Doc links in the commit message (b68ac26f4da8).

**Pull request merged**